src/compiler/binder.ts

```diff
@@ -2357,24 +2371,27 @@ namespace ts {
             bindPropertyAssignment(constructorFunction.escapedText, leftSideOfAssignment, /*isPrototypeProperty*/ true);
         }
 
-        function bindStaticPropertyAssignment(node: BinaryExpression) {
-            // We saw a node of the form 'x.y = z'. Declare a 'member' y on x if x was a function.
-
-            // Look up the function in the local scope, since prototype assignments should
+        /**
+         * For nodes like `x.y = z`, declare a member 'y' on 'x' if x was a function.

```

Comment:
Is the `if x was a function` qualifier in this comment still true?

Reply:
Updated: It's now function, class or not declared. I have a change that will make it also `{}`, but that's in a branch that I'll make a PR for after this PR is in.